services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -261,7 +264,17 @@ private UUID extractUUIDFromExistingManifestListPath(
    */
   private boolean doesPathExist(java.nio.file.Path tableDirPath) {
     try {
-      return fsStorageProvider.storageClient().exists(new Path(tableDirPath.toString()));
+      // TODO: Refactor client interaction to use high-level Storage API such as
+      // StorageManager::doesObjectExist
+      if (storageManager.getDefaultStorage().getType().equals(StorageType.HDFS)
+          || storageManager.getDefaultStorage().getType().equals(StorageType.LOCAL)) {
+        FileSystem fs =
+            (FileSystem) storageManager.getDefaultStorage().getClient().getNativeClient();
+        return fs.exists(new Path(tableDirPath.toString()));
+      } else {
+        throw new UnsupportedOperationException(
+            "Unsupported storage type for CTAS: " + storageManager.getDefaultStorage().getType());
```

Comment:
Why is a function doesPathExist related to SQL statement CTAS?

Reply:
This codepath mostly gets used in CTAS, and maybe for replication. But good point on not linking it to SQL.